sktime/datatypes/_convert.py

```diff
@@ -68,29 +68,37 @@
 
 import numpy as np
 import pandas as pd
+from typing import Union, List
 
 from sktime.datatypes._series import convert_dict_Series
 
 from sktime.datatypes._panel import convert_dict_Panel
 
 from sktime.datatypes._check import mtype as infer_mtype
 from sktime.datatypes._registry import mtype_to_scitype
+from sktime.datatypes.types import Mtypes, Scitypes
+
 
 # pool convert_dict-s and infer_mtype_dict-s
 convert_dict = dict()
 convert_dict.update(convert_dict_Series)
 convert_dict.update(convert_dict_Panel)
 
 
-def convert(obj, from_type: str, to_type: str, as_scitype: str = None, store=None):
+def convert(
+    obj, from_type: Mtypes, to_type: Mtypes, as_scitype: Scitypes = None, store=None
```

Comment:
I'd call it `Mtype` or `Scitype`, that makes more sense in the type hint?

Reply:
So I called it with the plural because the type itself is multiple things
```
Mtypes = Union[str, SeriesMtype, PanelMtype]
Scitypes = Union[str, Scitype]
```
Happy to change it to your suggestion however.

Reply:
aren't most types inhabited by multipe values? It's called `float` and not `floats`, `string` and not `strings`?

Reply:
that is a very good point :joy: